deps/build.jl

```diff
@@ -3,90 +3,41 @@ using BinaryProvider # requires BinaryProvider 0.3.0 or later
 # Parse some basic command-line arguments
 const verbose = "--verbose" in ARGS
 const prefix = Prefix(get([a for a in ARGS if a != "--verbose"], 1, joinpath(@__DIR__, "usr")))
+products = [
+    # LibraryProduct(prefix, ["libLLVM"], :libLLVM),
+    # LibraryProduct(prefix, ["libLTO"], :libLTO),
+    LibraryProduct(prefix, ["libclang"], :libclang),
+    # FileProduct(prefix, "tools/llvm-config", :llvm_config),
+]
 
-@static if Sys.isapple()
-    # download binaries from LLVM
-    bin_prefix = "http://releases.llvm.org/6.0.0"
-    url = "$bin_prefix/clang+llvm-6.0.0-x86_64-apple-darwin.tar.xz"
-    tarball_hash = "0ef8e99e9c9b262a53ab8f2821e2391d041615dd3f3ff36fdf5370916b0f4268"
-    # install
-    mac_prefix = Prefix(joinpath(prefix, "clang+llvm-6.0.0-x86_64-apple-darwin"))
-    products = [
-        LibraryProduct(mac_prefix, String["libclang"], :libclang),
-        ExecutableProduct(mac_prefix, "llvm-config", :llvm_config),
-    ]
-    install(url, tarball_hash; prefix=prefix, force=true, verbose=verbose)
-    # write out a deps.jl file that will contain mappings for our products
-    write_deps_file(joinpath(@__DIR__, "deps.jl"), products)
-end
-
-@static if Sys.islinux()
-    if !haskey(ENV, "LLVM_CONFIG")
-        @info "LLVM_CONFIG is not defined, looking for system llvm-config"
-        try
-            llvmcfg = readchomp(`which llvm-config`)
-            isfile(llvmcfg) && (ENV["LLVM_CONFIG"] = llvmcfg;)
-        catch err
-            error("no system llvm-config: $err")
-        end
-    end
-    LLVM_CONFIG = ENV["LLVM_CONFIG"]
-    llvm_libdir = readchomp(`$(LLVM_CONFIG) --libdir`)
-    if !ispath(llvm_libdir)
-        error("llvm-config returned non-existent libdir path: '$(llvm_libdir)")
-    end
+# Download binaries from hosted location
+bin_prefix = "https://github.com/ihnorton/Clang.jl/releases/download/Julia1.0-compatible"
```

Comment:
Cool!  This looks like we actually want to have a "LLVMBuilder.jl" that has a hard-bound on the Julia version that you can depend on through the package-system, so that each version of Julia get's the right version of LLVMBuilder (and of course we should provide compatible releases (my apologies)). Also it means  we should become a bit more conservative in what LLVMBuilder provides and maybe have a second mode for the more experimental features.



cc @staticfloat

Reply:
It saddens me that LLVM is so incompatible.

Let's create a `JuliaCompatibleLLVM` in [Yggdrasil](https://github.com/JuliaPackaging/Yggdrasil), register `JuliaCompatibleLLVM.jl`, hook up the appropriate `BinaryProvider` junk, and keep `JuliaCompatibleLLVM` as close to base Julia's `LLVM` as possible. We can keep `LLVMBuilder` with all the bells and whistles that others may want.